packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/endpoint_client.dart

```diff
@@ -48,35 +46,29 @@ class EndpointClient {
   /// {@macro amplify_analytics_pinpoint_dart.endpoint_client}
   @visibleForTesting
   Future<void> init({
-    required SecureStorageInterface endpointInfoStore,
-    LegacyNativeDataProvider? legacyNativeDataProvider,
+    required EndpointInfoStoreManager endpointInfoStoreManager,
   }) async {
-    final endpointIdManager = EndpointIdManager(
-      store: endpointInfoStore,
-      legacyNativeDataProvider: legacyNativeDataProvider,
+    _fixedEndpointId = await endpointInfoStoreManager.retrieveEndpointId(
+      pinpointAppId: _pinpointAppId,
     );
-    _fixedEndpointId = await endpointIdManager.retrieveEndpointId();
 
-    _globalFieldsManager =
-        await EndpointGlobalFieldsManager.create(endpointInfoStore);
+    _globalFieldsManager = await endpointInfoStoreManager.endpointFields;
```

Comment:
Inject this

Reply:
Please see my thoughts here: https://github.com/aws-amplify/amplify-flutter/pull/2639#discussion_r1102854770